File: 2025-1/sesion-de-clase-18/SoftProg/SoftProgNegocio/src/main/java/pe/edu/pucp/inf30/softprog/boimpl/CuentaUsuarioBOImpl.java
package pe.edu.pucp.inf30.softprog.boimpl;

import java.util.List;
import pe.edu.pucp.inf30.softprog.bo.Estado;
import pe.edu.pucp.inf30.softprog.bo.ICuentaUsuarioBO;
import pe.edu.pucp.inf30.softprog.dao.rrhh.ICuentaUsuarioDAO;
import pe.edu.pucp.inf30.softprog.daoimpl.rrhh.CuentaUsuarioDAOImpl;
import pe.edu.pucp.inf30.softprog.model.rrhh.CuentaUsuario;

/**
 *
 * @author eric
 */
public class CuentaUsuarioBOImpl implements ICuentaUsuarioBO {
    private final ICuentaUsuarioDAO cuentaUsuarioDao;
    
    public CuentaUsuarioBOImpl() {
        this.cuentaUsuarioDao = new CuentaUsuarioDAOImpl();
    }
    
    @Override
    public List<CuentaUsuario> listar() {
        return this.cuentaUsuarioDao.listar();
    }

    @Override
    public CuentaUsuario obtener(int id) {
        return this.cuentaUsuarioDao.buscar(id);
    }

    @Override
    public void eliminar(int id) {
        this.cuentaUsuarioDao.eliminar(id);
    }

    @Override
    public void guardar(CuentaUsuario modelo, Estado estado) {
        if (estado == Estado.Nuevo) {
            this.cuentaUsuarioDao.insertar(modelo);
        }
        else {
            this.cuentaUsuarioDao.modificar(modelo);
        }
    }
}

File: 2025-1/sesion-de-clase-23/SoftProgNet/SoftProgNegocio/BOImpl/CuentaUsuarioBOImpl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http.Headers;
using System.Security.Policy;
using System.Text;
using System.Threading.Tasks;
using PUCP.Edu.Pe.SoftProg.Modelo.RRHH;
using PUCP.Edu.Pe.SoftProg.Negocio.BO;
using PUCP.Edu.Pe.SoftProg.Persistencia.DAO.RRHH;
using PUCP.Edu.Pe.SoftProg.Persistencia.DAOImpl.RRHH;

namespace PUCP.Edu.Pe.SoftProg.Negocio.BOImpl {
    public class CuentaUsuarioBOImpl : ICuentaUsuarioBO {
        private ICuentaUsuarioDAO cuentausuarioDAO;

        public CuentaUsuarioBOImpl() {
            this.cuentausuarioDAO = new CuentaUsuarioDAOImpl();
        }

        public void Eliminar(int id) {
            this.cuentausuarioDAO.Eliminar(id);
        }

        public void Guardar(CuentaUsuario cuentaUsuario, Estado estado) {
            if (estado == Estado.Nuevo) {
                this.cuentausuarioDAO.Insertar(cuentaUsuario);
            }
            else if (estado == Estado.Modificar) {
                this.cuentausuarioDAO.Modificar(cuentaUsuario);
            }
        }

        public List<CuentaUsuario> Listar() {
            return this.cuentausuarioDAO.Listar();
        }

        public CuentaUsuario Obtener(int id) {
            return this.cuentausuarioDAO.Buscar(id);
        }
    }
}
